homeassistant/components/roborock/image.py

```diff
@@ -0,0 +1,130 @@
+"""Support for Roborock image."""
+import asyncio
+import io
+from itertools import chain
+
+from roborock import MultiMapsList, RoborockCommand
+from vacuum_map_parser_base.config.color import ColorsPalette
+from vacuum_map_parser_base.config.drawable import Drawable
+from vacuum_map_parser_base.config.image_config import ImageConfig
+from vacuum_map_parser_base.config.size import Sizes
+from vacuum_map_parser_roborock.map_data_parser import RoborockMapDataParser
+
+from homeassistant.components.image import ImageEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util import slugify
+import homeassistant.util.dt as dt_util
+
+from .const import DOMAIN
+from .coordinator import RoborockDataUpdateCoordinator
+from .device import RoborockCoordinatedEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Roborock image platform."""
+
+    coordinators: dict[str, RoborockDataUpdateCoordinator] = hass.data[DOMAIN][
+        config_entry.entry_id
+    ]
+    entities = list(
+        chain.from_iterable(
+            await asyncio.gather(
+                *(create_coordinator_maps(coord) for coord in coordinators.values())
+            )
+        )
+    )
+    async_add_entities(entities)
+
+
+class RoborockMap(RoborockCoordinatedEntity, ImageEntity):
+    """A class to let you visualize the map."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        unique_id: str,
+        coordinator: RoborockDataUpdateCoordinator,
+        map_flag: int,
+        starting_map: bytes,
+        map_name: str,
+    ) -> None:
+        """Initialize a Roborock map."""
+        RoborockCoordinatedEntity.__init__(self, unique_id, coordinator)
+        ImageEntity.__init__(self, coordinator.hass)
+        self._attr_name = map_name
+        drawables: list[Drawable] = [
+            Drawable.PATH,
+            Drawable.CHARGER,
+            Drawable.ROOM_NAMES,
+            Drawable.VACUUM_POSITION,
+        ]
+        self.parser = RoborockMapDataParser(
+            ColorsPalette(), Sizes(), drawables, ImageConfig(), []
+        )
+        self.async_update_token()
+        self._attr_image_last_updated = dt_util.utcnow()
+        self.map_flag = map_flag
+        self.cached_map = self._create_image(starting_map)
+
+    def should_update(self) -> bool:
```

Comment:
Given the name "update" has a specific name for entities and this uses an update coordinator, would a name like `is_cache_expired` be more accurate?